component/common/src/main/java/org/wso2/carbon/identity/smsotp/common/SMSOTPServiceImpl.java

```diff
@@ -23,6 +23,7 @@
 import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.log4j.MDC;
```

Comment:
This is log4j 1, please move this to log4j 2 based MDC

Reply:
instead we can use `org.apache.logging.log4j.ThreadContext;`

Reply:
I will use `threadcontext` in this case. However, I went with MDC in log4j1 as I came across [identity-api-user](https://github.com/wso2-support/identity-api-server/blob/support-1.0.82.x-full/components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/Util.java#L55) where the MDC in log4j1 is used. Are we gonna change here as well? or what shall we do? any thoughts?

Reply:
The MDC usages in the identity-api-user should be changed too.